pkg/certificate/mrc_reconciler.go

```diff
@@ -164,13 +167,19 @@ func (m *Manager) getCertIssuer(mrc *v1alpha2.MeshRootCertificate) (*issuer, err
 	return c, nil
 }
 
-func filterOutInactiveMRCs(mrcList []*v1alpha2.MeshRootCertificate) []*v1alpha2.MeshRootCertificate {
+func filterOutInactiveMRCs(mrcList []*v1alpha2.MeshRootCertificate) ([]*v1alpha2.MeshRootCertificate, error) {
 	n := 0
 	for _, mrc := range mrcList {
 		if mrc.Spec.Intent != v1alpha2.InactiveIntent {
 			mrcList[n] = mrc
 			n++
 		}
 	}
-	return mrcList[:n]
+
+	//Check if there more than 2 Active MRCs
+	if n > 2 {
```

Comment:
This check is already done later section. Is there a need to do it here?

Reply:
Sean suggested to add it in the last PR review. I can leave it as is or take it out. It does make some sense if we wanted to re-use this function and be safe.

Reply:
Reading Sean's comment, https://github.com/openservicemesh/osm/pull/5201/files#r1015697995, I don't think this code is protecting against 2 active MRCs and instead this is simply removing ones we don't care about.  Maybe @jaellio can comment?

Personally, I would lean towards avoiding it since this is an internal function and not part of an external api, my concern would be having the logic in two places from a maintenance perspective. This function will likely change or be removed in  https://github.com/openservicemesh/osm/issues/5226

Reply:
Yep, I agree. I don't think the check is necessary here.